migrations/2020/Version20200411105206.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Ramsey\Uuid\Uuid;
+use Throwable;
+
+final class Version20200411105206 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return 'Update cashbook int IDs to UUIDs';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        $this->addSql('SELECT 1');
+        $cashbookIds = $this->connection->fetchAll('SELECT id FROM `ac_object` WHERE length(id) < 36');
+        $this->connection->beginTransaction();
+        try {
+            foreach ($cashbookIds as $cashbookId) {
+                $ids = [Uuid::uuid4()->toString(), $cashbookId['id']];
+
+                $this->connection->executeUpdate('UPDATE ac_object         SET id = ?          WHERE id = ?', $ids);
+                $this->connection->executeUpdate('UPDATE ac_cashbook       SET id = ?          WHERE id = ?', $ids);
+                $this->connection->executeUpdate('UPDATE ac_chits          SET eventId = ?     WHERE eventId = ?', $ids);
+                $this->connection->executeUpdate('UPDATE ac_unit_cashbooks SET cashbook_id = ? WHERE cashbook_id = ?', $ids);
+            }
+            $this->connection->commit();
+        } catch (Throwable $exc) {
+            $this->connection->rollBack();
+        }
+    }
+
+    public function down(Schema $schema) : void
+    {
+        $this->addSql('SELECT 1');
```

Comment:
Tohle proč?

Reply:
Bez toho doctrine hlasi ze nebyla provedena zadna sql akce. Takze suppress hlaseni z doctrine

Reply:
V Down nám to nevadí a v up to jde pořešit tím, že nebudeme přímo pouštět SQLka, ale předáme je právě přes addSql()

Reply:
ok, upravil jsem to. Jeste jsem cetl, ze migrace jsou by default v transakci, tak jestli nemám hodit pryc tu transakci?

Reply:
Určitě ne, protože migrace jsou v transakci jen pokud daná databáze podporuje DDL commandy v transakci (bavili jsme se o tom včera v kontrastu k PostgreSQL)

Reply:
Ještě tenhle zbytečnej dotaz

Reply:
U toho down to bude jinak vyhazovat ten warning, coz nechci. Naopak to asi vrátím i do up, protoze kdyz nemáš žádný data v db , tak by se nikdy addSql nevolalo.